### PR TITLE
Patch ounit 2.0.0 to support the OCaml 4.06.0 bytes/string distinction

### PR DESCRIPTION
**Pull request by yallop**
/cc @gildor478

### CODE REVIEW

**Comment by yallop** [comment]
Upstream report: https://forge.ocamlcore.org/tracker/index.php?func=detail&aid=1760&group_id=162&atid=730

File: packages/ounit/ounit.2.0.0/opam [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+bug-reports: "https://forge.ocamlcore.org/tracker/?func=browse&group_id=162&atid=730"
+dev-repo: "https://github.com/gildor478/ounit.git"
 doc: ["http://ounit.forge.ocamlcore.org/api-ounit/index.html"]
 build: [make "build"]
 remove: [["ocamlfind" "remove" "oUnit"]]
 depends: [
   "ocamlfind"
   "ocamlbuild" {build}
 ]
+patches: ["safe-string.patch" {ocaml-version > "4.05.0"}]
```
[the remark this answers is not in the document]
Comment (yallop) [reply packages/ounit/ounit.2.0.0/opam]: Yes; that seems reasonable.  Done in 1e3c227.

File: packages/ounit/ounit.2.0.0/files/safe-string.patch [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+              let chn = open_in fn_out in
+-             let buff = String.make 4096 'X' in
++             let buff = Bytes.make 4096 'X' in
+              let len = ref (-1) in
+                while !len <> 0 do
+-                 len := input chn buff 0 (String.length buff);
++                 len := input chn buff 0 (Bytes.length buff);
+                  OUnitLogger.Test.raw_printf
+-                   ctxt.test_logger "%s" (String.sub buff 0 !len);
++                   ctxt.test_logger "%s" (Bytes.to_string (Bytes.sub buff 0 !len));
```
Comment (gasche) [inline packages/ounit/ounit.2.0.0/files/safe-string.patch]: You could use `Bytes.sub_string` here.

**Comment by gasche** [comment]
In the interest of helping the 4.06 release be as smooth as possible, I really appreciate the patching help (thanks!) and decided to go ahead and merge to let people test further packages. I'm still interested in any feedback/discussion on this PR, and we can always improve the patch afterwards.

**Comment by gildor478** [comment]
New release #10377

**Pull request merged**